src/telemetry.js

```diff
@@ -0,0 +1,100 @@
+var _ = require('./utility');
+
+function Telemeter(options) {
+  this.queue = [];
+  this.options = _.extend(true, {}, options);
+  var maxTelemetryEvents = this.options.maxTelemetryEvents;
+  // pin the queue size to the interval [0, 100]
+  this.maxQueueSize = Math.max(0, Math.min(maxTelemetryEvents || 100, 100))
+}
+
+Telemeter.prototype.copyEvents = function() {
+  return Array.prototype.slice.call(this.queue, 0);
+};
+
+Telemeter.prototype.capture = function(type, metadata, level) {
+  var e = {
+    level: getLevel(type, level),
+    type: type,
+    timestamp_ms: now(),
+    metadata: metadata
+  };
+  this.push(e);
+  return e;
+};
+
+Telemeter.prototype.captureError = function(err) {
```

Comment:
It's a bit weird to have this as well as `captureRollbar` but I can see a reason to have a distinction. What if we added an optional Rollbar param to each of these methods to link them with a Rollbar occurrence if one was sent for the event?

E.g.

```js
Telemeter.prototype.captureError = function(err, rollbar_uuid=null) {
  //...
}
Telemeter.prototype.captureLog = function(err, rollbar_uuid=null) {
  //...
}

//...
```

This way we allow the addition of telemetry items which are and are not associated with Rollbar occurrences.